src/v/features/feature_table.cc

```diff
@@ -446,6 +446,20 @@ void feature_table::testing_activate_all() {
     on_update();
 }
 
+feature_table::version_fence
+feature_table::decode_version_fence(model::record_batch batch) {
+    auto records = batch.copy_records();
+    if (records.empty()) {
+        throw std::runtime_error("");
```

Comment:
nit: could you make these throw messages clearer so we can at least identify what's wrong with the batch in case there are serialization issues?